doc/build_system.dox

```diff
@@ -0,0 +1,549 @@
+/**
+@page build_system_page Build system
+
+@tableofcontents
+
+@section Overview
+
+The key added value of the OpenMRN build system is that it allows compiling a
+multitude of applications across vastly different hardware architectures from a
+single set of source files. It allows selecting different toolchains (e.g. g++
+or clang for linux, mingw for windows, armgcc for cortex-m3, emscripten-clang
+for node.js), providing different compilation options and selecting different
+set of source files to compile on each specific port. Applications can share
+operating system support files, drivers, networking stack, system libraries, or
+libraries defining application logic.
+
+@section core_concepts Concepts
+
+The OpenMRN build system is based upon recursive make. Each directory contains
+a Makefile, and during the compilation process these makefiles instruct the
+build program to recurse into the relevant subdirectories and execute `make`
+there.
+
+The most important concept is the *target*.
+- It defines the toolchain and compiler settings to be used;
+- defines which operating system the code shall be compiled for;
+- sets all the compilation flags;
+- defines the code that shall be compiled vs skipped.
+
+Generally, all code in `openmrn/src` gets compiled once per *core target*, and
+all code in the application tree gets compiled once per *application target*.
```

Comment:
you can make *core target* and *application target* @ref hyperlinks, e.g. @ref core_target